### PR TITLE
NavBar docs

### PR DESCRIPTION
**Pull request by ChrisRzech**
### Key features
* Documentation for `NavBar`

### Future Improvements
* More in-depth documentation

### Notes
none

### CODE REVIEW

File: src/widgets/navbar.cpp [hunk 1]
```diff
@@ -2,7 +2,16 @@
 #include "navitem.hpp"
 #include <QPropertyAnimation>
 
-/* Constructor */
+/**
+ * @brief Constructor
```
Comment (typokign) [inline src/widgets/navbar.cpp:6]: useless brief

File: src/widgets/navbar.cpp [hunk 2]
```diff
@@ -25,23 +34,54 @@ NavBar::NavBar(QWidget* parent, int minWidth, int maxWidth)
 
 }
 
-/* Setters */
+/**
+ * @brief Set height of the bar
```
Comment (typokign) [inline src/widgets/navbar.cpp:38]: really no need to document setters and getters unless there's some really weird pre-condition or side effect. but even in those cases it probably just means you need to refactor

File: src/widgets/navbar.cpp [hunk 3]
```diff
@@ -25,23 +34,54 @@ NavBar::NavBar(QWidget* parent, int minWidth, int maxWidth)
...
+/**
+ * @brief Add button to the bar
+ *
+ * Adds a new button to the bar using the given strings.
+ * Since NavBar uses "Font Awesome 5 Free", all unicode
+ * string icons should be represented by that font.
+ *
+ * @param icon The unicode representation of the icon in the font "Font Awesome 5 Free"
+ * @param label The label of the button
+ */
```
Comment (typokign) [inline src/widgets/navbar.cpp:84]: now this is a good docstring, but the `icon` parameter could definitely use an example. By Unicode representation do you mean just the codepoint? Hex or dec? Or `\x[codept]`? Or `\u[codept]`?

File: src/widgets/navbar.hpp [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+ * @brief NavBar class
+ *
+ * Provides an easy way to list clickable buttons.
+ * Currently, the class uses "Font Awesome 5 Free" for its icons.
+ *
+ * To use:
+ * @code{.cpp}
+ * NavBar* navbar = new NavBar(widget, 50, 100);
+ * navbar->addItems("\uf0ca", "Button");
+ * @endcode
```
Comment (typokign) [inline src/widgets/navbar.hpp:14]: good example

File: src/widgets/navbar.cpp [hunk 5]
```diff
@@ -25,23 +34,54 @@ NavBar::NavBar(QWidget* parent, int minWidth, int maxWidth)
...
 void NavBar::setMaxWidth(int width)
 {
     m_maxWidth = width;
 }
 
-/* List modifiers */
+/**
+ * @brief Add button to the bar
+ *
+ * Adds a new button to the bar using the given strings.
```
Comment (typokign) [inline src/widgets/navbar.cpp:78]: Would be nice to add this somewhere:

`@see <a href="https://fontawesome.com/free">Font Awesome icon reference</a>`

**Review by Roger-Chavez — APPROVED** [review]

**Pull request merged**